lib/puppet/provider/group/windows_adsi.rb

```diff
@@ -22,9 +22,6 @@ def members_insync?(current, should)
 
     # Cannot use munge of the group property to canonicalize @should
     # since the default array_matching comparison is not commutative
-    should_empty = should.nil? or should.empty?
-
-    return false if current.empty? != should_empty
```

Comment:
The guy who worked on https://github.com/puppetlabs/puppet/commit/455834e7f21f213e9d0085510d5b47c0c3e2a1ab is a total :monkey: